<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
    "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta http-equiv="Content-Type" content="application/xhtml+xml; charset=UTF-8" />
<meta name="generator" content="AsciiDoc 8.6.8" />
<title></title>
<style type="text/css">
/* Shared CSS for AsciiDoc xhtml11 and html5 backends */

/* Default font. */
body {
  font-family: Georgia,serif;
}

/* Title font. */
h1, h2, h3, h4, h5, h6,
div.title, caption.title,
thead, p.table.header,
#toctitle,
#author, #revnumber, #revdate, #revremark,
#footer {
  font-family: Arial,Helvetica,sans-serif;
}

body {
  margin: 1em 5% 1em 5%;
}

a {
  color: blue;
  text-decoration: underline;
}
a:visited {
  color: fuchsia;
}

em {
  font-style: italic;
  color: navy;
}

strong {
  font-weight: bold;
  color: #083194;
}

h1, h2, h3, h4, h5, h6 {
  color: #527bbd;
  margin-top: 1.2em;
  margin-bottom: 0.5em;
  line-height: 1.3;
}

h1, h2, h3 {
  border-bottom: 2px solid silver;
}
h2 {
  padding-top: 0.5em;
}
h3 {
  float: left;
}
h3 + * {
  clear: left;
}
h5 {
  font-size: 1.0em;
}

div.sectionbody {
  margin-left: 0;
}

hr {
  border: 1px solid silver;
}

p {
  margin-top: 0.5em;
  margin-bottom: 0.5em;
}

ul, ol, li > p {
  margin-top: 0;
}
ul > li     { color: #aaa; }
ul > li > * { color: black; }

.monospaced, code, pre {
  font-family: "Courier New", Courier, monospace;
  font-size: inherit;
  color: navy;
  padding: 0;
  margin: 0;
}


#author {
  color: #527bbd;
  font-weight: bold;
  font-size: 1.1em;
}
#email {
}
#revnumber, #revdate, #revremark {
}

#footer {
  font-size: small;
  border-top: 2px solid silver;
  padding-top: 0.5em;
  margin-top: 4.0em;
}
#footer-text {
  float: left;
  padding-bottom: 0.5em;
}
#footer-badges {
  float: right;
  padding-bottom: 0.5em;
}

#preamble {
  margin-top: 1.5em;
  margin-bottom: 1.5em;
}
div.imageblock, div.exampleblock, div.verseblock,
div.quoteblock, div.literalblock, div.listingblock, div.sidebarblock,
div.admonitionblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
div.admonitionblock {
  margin-top: 2.0em;
  margin-bottom: 2.0em;
  margin-right: 10%;
  color: #606060;
}

div.content { /* Block element content. */
  padding: 0;
}

/* Block element titles. */
div.title, caption.title {
  color: #527bbd;
  font-weight: bold;
  text-align: left;
  margin-top: 1.0em;
  margin-bottom: 0.5em;
}
div.title + * {
  margin-top: 0;
}

td div.title:first-child {
  margin-top: 0.0em;
}
div.content div.title:first-child {
  margin-top: 0.0em;
}
div.content + div.title {
  margin-top: 0.0em;
}

div.sidebarblock > div.content {
  background: #ffffee;
  border: 1px solid #dddddd;
  border-left: 4px solid #f0f0f0;
  padding: 0.5em;
}

div.listingblock > div.content {
  border: 1px solid #dddddd;
  border-left: 5px solid #f0f0f0;
  background: #f8f8f8;
  padding: 0.5em;
}

div.quoteblock, div.verseblock {
  padding-left: 1.0em;
  margin-left: 1.0em;
  margin-right: 10%;
  border-left: 5px solid #f0f0f0;
  color: #888;
}

div.quoteblock > div.attribution {
  padding-top: 0.5em;
  text-align: right;
}

div.verseblock > pre.content {
  font-family: inherit;
  font-size: inherit;
}
div.verseblock > div.attribution {
  padding-top: 0.75em;
  text-align: left;
}
/* DEPRECATED: Pre version 8.2.7 verse style literal block. */
div.verseblock + div.attribution {
  text-align: left;
}

div.admonitionblock .icon {
  vertical-align: top;
  font-size: 1.1em;
  font-weight: bold;
  text-decoration: underline;
  color: #527bbd;
  padding-right: 0.5em;
}
div.admonitionblock td.content {
  padding-left: 0.5em;
  border-left: 3px solid #dddddd;
}

div.exampleblock > div.content {
  border-left: 3px solid #dddddd;
  padding-left: 0.5em;
}

div.imageblock div.content { padding-left: 0; }
span.image img { border-style: none; }
a.image:visited { color: white; }

dl {
  margin-top: 0.8em;
  margin-bottom: 0.8em;
}
dt {
  margin-top: 0.5em;
  margin-bottom: 0;
  font-style: normal;
  color: navy;
}
dd > *:first-child {
  margin-top: 0.1em;
}

ul, ol {
    list-style-position: outside;
}
ol.arabic {
  list-style-type: decimal;
}
ol.loweralpha {
  list-style-type: lower-alpha;
}
ol.upperalpha {
  list-style-type: upper-alpha;
}
ol.lowerroman {
  list-style-type: lower-roman;
}
ol.upperroman {
  list-style-type: upper-roman;
}

div.compact ul, div.compact ol,
div.compact p, div.compact p,
div.compact div, div.compact div {
  margin-top: 0.1em;
  margin-bottom: 0.1em;
}

tfoot {
  font-weight: bold;
}
td > div.verse {
  white-space: pre;
}

div.hdlist {
  margin-top: 0.8em;
  margin-bottom: 0.8em;
}
div.hdlist tr {
  padding-bottom: 15px;
}
dt.hdlist1.strong, td.hdlist1.strong {
  font-weight: bold;
}
td.hdlist1 {
  vertical-align: top;
  font-style: normal;
  padding-right: 0.8em;
  color: navy;
}
td.hdlist2 {
  vertical-align: top;
}
div.hdlist.compact tr {
  margin: 0;
  padding-bottom: 0;
}

.comment {
  background: yellow;
}

.footnote, .footnoteref {
  font-size: 0.8em;
}

span.footnote, span.footnoteref {
  vertical-align: super;
}

#footnotes {
  margin: 20px 0 20px 0;
  padding: 7px 0 0 0;
}

#footnotes div.footnote {
  margin: 0 0 5px 0;
}

#footnotes hr {
  border: none;
  border-top: 1px solid silver;
  height: 1px;
  text-align: left;
  margin-left: 0;
  width: 20%;
  min-width: 100px;
}

div.colist td {
  padding-right: 0.5em;
  padding-bottom: 0.3em;
  vertical-align: top;
}
div.colist td img {
  margin-top: 0.3em;
}

@media print {
  #footer-badges { display: none; }
}

#toc {
  margin-bottom: 2.5em;
}

#toctitle {
  color: #527bbd;
  font-size: 1.1em;
  font-weight: bold;
  margin-top: 1.0em;
  margin-bottom: 0.1em;
}

div.toclevel0, div.toclevel1, div.toclevel2, div.toclevel3, div.toclevel4 {
  margin-top: 0;
  margin-bottom: 0;
}
div.toclevel2 {
  margin-left: 2em;
  font-size: 0.9em;
}
div.toclevel3 {
  margin-left: 4em;
  font-size: 0.9em;
}
div.toclevel4 {
  margin-left: 6em;
  font-size: 0.9em;
}

span.aqua { color: aqua; }
span.black { color: black; }
span.blue { color: blue; }
span.fuchsia { color: fuchsia; }
span.gray { color: gray; }
span.green { color: green; }
span.lime { color: lime; }
span.maroon { color: maroon; }
span.navy { color: navy; }
span.olive { color: olive; }
span.purple { color: purple; }
span.red { color: red; }
span.silver { color: silver; }
span.teal { color: teal; }
span.white { color: white; }
span.yellow { color: yellow; }

span.aqua-background { background: aqua; }
span.black-background { background: black; }
span.blue-background { background: blue; }
span.fuchsia-background { background: fuchsia; }
span.gray-background { background: gray; }
span.green-background { background: green; }
span.lime-background { background: lime; }
span.maroon-background { background: maroon; }
span.navy-background { background: navy; }
span.olive-background { background: olive; }
span.purple-background { background: purple; }
span.red-background { background: red; }
span.silver-background { background: silver; }
span.teal-background { background: teal; }
span.white-background { background: white; }
span.yellow-background { background: yellow; }

span.big { font-size: 2em; }
span.small { font-size: 0.6em; }

span.underline { text-decoration: underline; }
span.overline { text-decoration: overline; }
span.line-through { text-decoration: line-through; }

div.unbreakable { page-break-inside: avoid; }


/*
 * xhtml11 specific
 *
 * */

div.tableblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
div.tableblock > table {
  border: 3px solid #527bbd;
}
thead, p.table.header {
  font-weight: bold;
  color: #527bbd;
}
p.table {
  margin-top: 0;
}
/* Because the table frame attribute is overriden by CSS in most browsers. */
div.tableblock > table[frame="void"] {
  border-style: none;
}
div.tableblock > table[frame="hsides"] {
  border-left-style: none;
  border-right-style: none;
}
div.tableblock > table[frame="vsides"] {
  border-top-style: none;
  border-bottom-style: none;
}


/*
 * html5 specific
 *
 * */

table.tableblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
thead, p.tableblock.header {
  font-weight: bold;
  color: #527bbd;
}
p.tableblock {
  margin-top: 0;
}
table.tableblock {
  border-width: 3px;
  border-spacing: 0px;
  border-style: solid;
  border-color: #527bbd;
  border-collapse: collapse;
}
th.tableblock, td.tableblock {
  border-width: 1px;
  padding: 4px;
  border-style: solid;
  border-color: #527bbd;
}

table.tableblock.frame-topbot {
  border-left-style: hidden;
  border-right-style: hidden;
}
table.tableblock.frame-sides {
  border-top-style: hidden;
  border-bottom-style: hidden;
}
table.tableblock.frame-none {
  border-style: hidden;
}

th.tableblock.halign-left, td.tableblock.halign-left {
  text-align: left;
}
th.tableblock.halign-center, td.tableblock.halign-center {
  text-align: center;
}
th.tableblock.halign-right, td.tableblock.halign-right {
  text-align: right;
}

th.tableblock.valign-top, td.tableblock.valign-top {
  vertical-align: top;
}
th.tableblock.valign-middle, td.tableblock.valign-middle {
  vertical-align: middle;
}
th.tableblock.valign-bottom, td.tableblock.valign-bottom {
  vertical-align: bottom;
}


/*
 * manpage specific
 *
 * */

body.manpage h1 {
  padding-top: 0.5em;
  padding-bottom: 0.5em;
  border-top: 2px solid silver;
  border-bottom: 2px solid silver;
}
body.manpage h2 {
  border-style: none;
}
body.manpage div.sectionbody {
  margin-left: 3em;
}

@media print {
  body.manpage div#toc { display: none; }
}


</style>
<script type="text/javascript">
/*<![CDATA[*/
var asciidoc = {  // Namespace.

/////////////////////////////////////////////////////////////////////
// Table Of Contents generator
/////////////////////////////////////////////////////////////////////

/* Author: Mihai Bazon, September 2002
 * http://students.infoiasi.ro/~mishoo
 *
 * Table Of Content generator
 * Version: 0.4
 *
 * Feel free to use this script under the terms of the GNU General Public
 * License, as long as you do not remove or alter this notice.
 */

 /* modified by Troy D. Hanson, September 2006. License: GPL */
 /* modified by Stuart Rackham, 2006, 2009. License: GPL */

// toclevels = 1..4.
toc: function (toclevels) {

  function getText(el) {
    var text = "";
    for (var i = el.firstChild; i != null; i = i.nextSibling) {
      if (i.nodeType == 3 /* Node.TEXT_NODE */) // IE doesn't speak constants.
        text += i.data;
      else if (i.firstChild != null)
        text += getText(i);
    }
    return text;
  }

  function TocEntry(el, text, toclevel) {
    this.element = el;
    this.text = text;
    this.toclevel = toclevel;
  }

  function tocEntries(el, toclevels) {
    var result = new Array;
    var re = new RegExp('[hH]([1-'+(toclevels+1)+'])');
    // Function that scans the DOM tree for header elements (the DOM2
    // nodeIterator API would be a better technique but not supported by all
    // browsers).
    var iterate = function (el) {
      for (var i = el.firstChild; i != null; i = i.nextSibling) {
        if (i.nodeType == 1 /* Node.ELEMENT_NODE */) {
          var mo = re.exec(i.tagName);
          if (mo && (i.getAttribute("class") || i.getAttribute("className")) != "float") {
            result[result.length] = new TocEntry(i, getText(i), mo[1]-1);
          }
          iterate(i);
        }
      }
    }
    iterate(el);
    return result;
  }

  var toc = document.getElementById("toc");
  if (!toc) {
    return;
  }

  // Delete existing TOC entries in case we're reloading the TOC.
  var tocEntriesToRemove = [];
  var i;
  for (i = 0; i < toc.childNodes.length; i++) {
    var entry = toc.childNodes[i];
    if (entry.nodeName.toLowerCase() == 'div'
     && entry.getAttribute("class")
     && entry.getAttribute("class").match(/^toclevel/))
      tocEntriesToRemove.push(entry);
  }
  for (i = 0; i < tocEntriesToRemove.length; i++) {
    toc.removeChild(tocEntriesToRemove[i]);
  }

  // Rebuild TOC entries.
  var entries = tocEntries(document.getElementById("content"), toclevels);
  for (var i = 0; i < entries.length; ++i) {
    var entry = entries[i];
    if (entry.element.id == "")
      entry.element.id = "_toc_" + i;
    var a = document.createElement("a");
    a.href = "#" + entry.element.id;
    a.appendChild(document.createTextNode(entry.text));
    var div = document.createElement("div");
    div.appendChild(a);
    div.className = "toclevel" + entry.toclevel;
    toc.appendChild(div);
  }
  if (entries.length == 0)
    toc.parentNode.removeChild(toc);
},


/////////////////////////////////////////////////////////////////////
// Footnotes generator
/////////////////////////////////////////////////////////////////////

/* Based on footnote generation code from:
 * http://www.brandspankingnew.net/archive/2005/07/format_footnote.html
 */

footnotes: function () {
  // Delete existing footnote entries in case we're reloading the footnodes.
  var i;
  var noteholder = document.getElementById("footnotes");
  if (!noteholder) {
    return;
  }
  var entriesToRemove = [];
  for (i = 0; i < noteholder.childNodes.length; i++) {
    var entry = noteholder.childNodes[i];
    if (entry.nodeName.toLowerCase() == 'div' && entry.getAttribute("class") == "footnote")
      entriesToRemove.push(entry);
  }
  for (i = 0; i < entriesToRemove.length; i++) {
    noteholder.removeChild(entriesToRemove[i]);
  }

  // Rebuild footnote entries.
  var cont = document.getElementById("content");
  var spans = cont.getElementsByTagName("span");
  var refs = {};
  var n = 0;
  for (i=0; i<spans.length; i++) {
    if (spans[i].className == "footnote") {
      n++;
      var note = spans[i].getAttribute("data-note");
      if (!note) {
        // Use [\s\S] in place of . so multi-line matches work.
        // Because JavaScript has no s (dotall) regex flag.
        note = spans[i].innerHTML.match(/\s*\[([\s\S]*)]\s*/)[1];
        spans[i].innerHTML =
          "[<a id='_footnoteref_" + n + "' href='#_footnote_" + n +
          "' title='View footnote' class='footnote'>" + n + "</a>]";
        spans[i].setAttribute("data-note", note);
      }
      noteholder.innerHTML +=
        "<div class='footnote' id='_footnote_" + n + "'>" +
        "<a href='#_footnoteref_" + n + "' title='Return to text'>" +
        n + "</a>. " + note + "</div>";
      var id =spans[i].getAttribute("id");
      if (id != null) refs["#"+id] = n;
    }
  }
  if (n == 0)
    noteholder.parentNode.removeChild(noteholder);
  else {
    // Process footnoterefs.
    for (i=0; i<spans.length; i++) {
      if (spans[i].className == "footnoteref") {
        var href = spans[i].getElementsByTagName("a")[0].getAttribute("href");
        href = href.match(/#.*/)[0];  // Because IE return full URL.
        n = refs[href];
        spans[i].innerHTML =
          "[<a href='#_footnote_" + n +
          "' title='View footnote' class='footnote'>" + n + "</a>]";
      }
    }
  }
},

install: function(toclevels) {
  var timerId;

  function reinstall() {
    asciidoc.footnotes();
    if (toclevels) {
      asciidoc.toc(toclevels);
    }
  }

  function reinstallAndRemoveTimer() {
    clearInterval(timerId);
    reinstall();
  }

  timerId = setInterval(reinstall, 500);
  if (document.addEventListener)
    document.addEventListener("DOMContentLoaded", reinstallAndRemoveTimer, false);
  else
    window.onload = reinstallAndRemoveTimer;
}

}
asciidoc.install();
/*]]>*/
</script>
</head>
<body class="article">
<div id="header">
</div>
<div id="content">
<div class="sect1">
<h2 id="_requirements">Requirements</h2>
<div class="sectionbody">
<div class="paragraph"><p>Ensure that each of the software tools mentioned in this section (except Trinity) are available within your unix PATH setting.  For example, if you have tools installed in a “/usr/local/tools” directory, you can update your PATH setting to include this directory in the search path by:
      %  export PATH=/usr/local/tools:$PATH</p></div>
<div class="ulist"><ul>
<li>
<p>
Hardware (64-bit computer running Linux; <sub>1G of RAM per </sub>1M paired-end reads)
</p>
</li>
<li>
<p>
Trinity version trinityrnaseq_r2013-02-25: <a href="http://trinityrnaseq.sourceforge.net">http://trinityrnaseq.sourceforge.net</a>
</p>
</li>
<li>
<p>
Bowtie version 0.12.9: <a href="http://bowtie-bio.sourceforge.net">http://bowtie-bio.sourceforge.net</a> (RSEM is currently not compatible with Bowtie 2, so be sure to obtain the latest release for Bowtie 1, which is currently v. 0.12.9, as released on December 16, 2012)
</p>
</li>
<li>
<p>
Samtools version 0.1.18: <a href="http://sourceforge.net/projects/samtools/files/samtools/">http://sourceforge.net/projects/samtools/files/samtools/</a>
</p>
</li>
<li>
<p>
R version 2.15: <a href="http://www.r-project.org">http://www.r-project.org</a>, requires packages: edgeR, ctc, Biobase, ape, and gplots
</p>
</li>
<li>
<p>
Blast+ version 2.2.27: <a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/">ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/</a>
</p>
</li>
</ul></div>
</div>
</div>
<div class="sect1">
<h2 id="_equipment_setup">Equipment Setup</h2>
<div class="sectionbody">
<div class="paragraph"><p>Optionally, and only for simplicity, define the environmental variable TRINITY_HOME, replacing “/software/trinityrnaseq” below with the path to your Trinity software installation. Alternatively, write the full path where $TRINITY_HOME appears in the Procedure</p></div>
<div class="literalblock">
<div class="content">
<pre><code>% export TRINITY_HOME=/software/trinityrnaseq</code></pre>
</div></div>
</div>
</div>
<div class="sect1">
<h2 id="_procedure">Procedure</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_collection_of_rna_seq_data_timing_10_minutes">Collection of RNA-Seq data Timing: ~10 minutes===</h3>
<div class="olist arabic"><ol class="arabic" start="1">
<li>
<p>
Download strand-specific RNA-Seq data from Schizosaccharomyces pombe grown in four conditions45 (logarithmic growth, plateau phase, diauxic shift, and heat shock, each with 1M Illumina paired-end strand-specific RNA-Seq data, to a total of 4M paired-end reads) by visiting the URL reported below in a web browser, or directly from the command line using ‘wget’:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% wget http://sourceforge.net/projects/trinityrnaseq/files/misc/TrinityNatureProtocolTutorial.tgz/download</code></pre>
</div></div>
</li>
<li>
<p>
Name the file thus downloaded ‘TrinityNatureProtocolTutorial.tgz’, which should be 540 MB in size.
</p>
</li>
<li>
<p>
Unpack this file by:
</p>
<div class="literalblock">
<div class="content">
<pre><code>tar –xvf TrinityNatureProtocolTutorial.tgz</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>which should generate the following files in a TrinityNatureProtocolTutorial/ directory with the following contents:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>S_pombe_refTrans.fasta   # reference transcriptome for S. pombe</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>1M_READS_sample/Sp.hs.1M.left.fq   # PE reads for heatshock
1M_READS_sample/Sp.hs.1M.right.fq</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>1M_READS_sample/Sp.log.1M.left.fq  # PE reads for log phase
1M_READS_sample/Sp.log.1M.right.fq</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>1M_READS_sample/Sp.ds.1M.right.fq   # PE reads for diauxic shock
1M_READS_sample/Sp.ds.1M.left.fq</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>1M_READS_sample/Sp.plat.1M.left.fq  # PE reads for plateau phase
1M_READS_sample/Sp.plat.1M.right.fq</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>samples_n_reads_described.txt   # tab-delimited description file.</code></pre>
</div></div>
<div class="paragraph"><p>Please note that this protocol expects the raw data to be of high quality, free from adaptor, barcodes and other contaminating subsequences.
De novo RNA-Seq assembly using Trinity Timing: 60–90 min</p></div>
<div class="olist arabic"><ol class="arabic" start="4">
<li>
<p>
Create a working folder and place the ‘TrinityNatureProtocolTutorial/’ directory contents there (as per the Materials section).
</p>
</li>
<li>
<p>
In order to facilitate downstream analyses, concatenate the RNA-Seq data across all samples into a single set of inputs to generate a single reference Trinity assembly. Combine all ‘left’ reads into a single file, and combine all ‘right’ reads into a single file by:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% cat 1M_READS_sample/*.left.fq &gt; reads.ALL.left.fq</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>% cat 1M_READS_sample/*.right.fq &gt; reads.ALL.right.fq</code></pre>
</div></div>
</li>
<li>
<p>
Assemble the reads into transcripts using Trinity by:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/Trinity.pl --seqType fq --JM 10G \
    --left reads.ALL.left.fq --right reads.ALL.right.fq \
    --SS_lib_type RF --CPU 6</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>Please note that the --JM option enables the user to control the amount of RAM used during Jellyfish k-mer counting, in this case, 10 Gb. The --CPU option controls the number of parallel processes. Feel free to change these parameters depending on your system. The Trinity reconstructed transcripts will exist as FASTA-formatted sequences in the output file ‘trinity_out_dir/Trinity.fasta’.</p></div>
<div class="olist arabic"><ol class="arabic" start="7">
<li>
<p>
Use the script ‘$TRINITY_HOME/utilities/TrinityStats.pl’ to examine the statistic for the Trinity assemblies:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/TrinityStats.pl trinity_out_dir/Trinity.fasta</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>The ‘TrinityStats.pl’ reports the number of transcripts, components, and the transcript contig N50 value based on the ‘Trinity.fasta’ file.  The contig N50 value, defined as the maximum length whereby at least 50% of the total assembled sequence resides in contigs of at least that length, is a commonly used metric for evaluating the contiguity of a genome assembly. Note that, unlike genome assemblies, maximizing N50 is not appropriate for transcriptomes; it is more appropriate to use an index based on a reference dataset (from the same or a closely related species) and to estimate the number of reference genes recovered and how many can be deemed to be full-length49, 50 (see below). The N50 value is, however, useful for confirming that the assembly succeeded (you will expect a value that is near the average transcript length of S. pombe , avg. = 1397 bases)</p></div>
</div>
<div class="sect2">
<h3 id="_quality_assessment_timing_90_min">Quality assessment Timing: ~90 min</h3>
<div class="admonitionblock">
<table><tr>
<td class="icon">
<div class="title">Note</div>
</td>
<td class="content">This section of the Procedure is optional, but we highly recommend its implementation</td>
</tr></table>
</div>
<div class="olist arabic"><ol class="arabic" start="8">
<li>
<p>
Examine the breadth of genetic composition and transcript contiguity by leveraging a reference data set. The annotated reference transcriptome of Schizosaccharomyces pombe is included as file ‘S_pombe_refTrans.fasta’. Use megablast and our included analysis script to analyze its representation by the Trinity assembly as described below (Steps 9-13).
</p>
</li>
<li>
<p>
Prepare the reference transcriptome fasta file as a BLAST database:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% makeblastdb -in S_pombe_refTrans.fasta -dbtype nucl</code></pre>
</div></div>
</li>
<li>
<p>
Run megablast to align the known transcripts to the Trinity assembly:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% blastn -query trinity_out_dir/Trinity.fasta \
    -db S_pombe_refTrans.fasta \
    -out Trinity_vs_S_pombe_refTrans.blastn \
    -evalue 1e-20 -dust no -task megablast -num_threads 2 \
    -max_target_seqs 1 -outfmt 6</code></pre>
</div></div>
</li>
<li>
<p>
Once megablast has completed, run the script below to examine the length coverage of top database hits:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/analyze_blastPlus_topHit_coverage.pl  \
          Trinity_vs_S_pombe_genes.blastn \
          trinity_out_dir/Trinity.fasta \
          S_pombe_refTrans.fasta</code></pre>
</div></div>
</li>
<li>
<p>
Examine the number of input RNA-Seq reads that are well represented by the transcriptome assembly.  Trinity provides a script ‘alignReads.pl’ that executes Bowtie to align the left and right fragment reads separately to the Trinity contigs and then groups the reads together into pairs while retaining those single read alignments that are not found to be properly paired with their mate. Run ‘alignReads.pl’ by:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/alignReads.pl --seqType fq \
                                   --left reads.ALL.left.fq --right reads.ALL.right.fq \
                                   --SS_lib_type RF --retain_intermediate_files \
                                   --aligner bowtie \
                                --target trinity_out_dir/Trinity.fasta -- -p 4</code></pre>
</div></div>
</li>
<li>
<p>
When ‘alignReads.pl’ is run using strand-specific data, as indicated above with the ‘--SS_lib_type RF’ parameter setting, it will separate the alignments into those that align to the sense strand (‘+’) from the antisense strand (‘-’).  All output files including coordinate-sorted and read-name-sorted SAM files should exist in a ‘bowtie_out/’ directory.  Count the number of reads aligning (at least once) to the sense strand of transcripts by running the utility below on the sense-strand read name-sorted alignment file as shown:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/SAM_nameSorted_to_uniq_count_stats.pl   \
                           bowtie_out/bowtie_out.nameSorted.sam.+.sam</code></pre>
</div></div>
</li>
</ol></div>
</div>
<div class="sect2">
<h3 id="_abundance_estimation_using_rsem_timing_40_60_min">Abundance estimation using RSEM Timing: 40–60 min.</h3>
<div class="olist arabic"><ol class="arabic" start="14">
<li>
<p>
Obtain transcript abundance estimates by running RSEM separately for each sample, as shown below:  The PERL script ‘run_RSEM_align_n_estimate.pl’ simply provides an interface to the RSEM software, translating the familiar Trinity command-line parameters to their RSEM equivalents and then executing the RSEM software. Each relevant step below (15–18) generates files ‘$prefix.isoforms.results’ and ‘$prefix.genes.results’, containing the abundance estimations for Trinity transcripts (Table 1) and components (Table 2), respectively.  The $prefix in the filename is set based on the ‘--prefix’ setting in the commands below, which is unique to each sample. Please note that, with regard to the parameters reported in Table 1 and Table 2, Transcript_id is the Trinity transcript identifier; Gene_id is the Trinity component to which the reconstructed transcript was derived; Length is the length of the reconstructed transcript; Effective length is the mean number of 5' start positions from which an RNA-Seq fragment could have been derived from this transcript, given the distribution of fragment lengths inferred by RSEM — the value is equal to (transcript_length - mean_fragment_length + 1); Expected count is the number of expected RNA-Seq fragments assigned to the transcript given maximum likelihood transcript abundance estimates; TPM is the number of transcripts per million; FPKM is the number of RNA-Seq fragments per kilobase of transcript effective length per million fragments mapped to all transcripts; and IsoPct is the percent of expression for a given transcript compared to all expression from that Trinity component. Note also that the gene&#8217;s length and effective length are defined as the IsoPct weighted sum of transcript lengths and effective lengths. The gene&#8217;s expected counts, TPM, and FPKM are defined as the sum of its transcripts' expected counts, TPM and FPKM.
</p>
</li>
<li>
<p>
RSEM for log phase
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/RSEM_util/run_RSEM_align_n_estimate.pl \
   --transcripts trinity_out_dir/Trinity.fasta \
   --left Sp.log.1M.left.fq \
   --right Sp.log.1M.right.fq \
   --seqType fq \
   --SS_lib_type RF \
   --prefix LOG</code></pre>
</div></div>
</li>
<li>
<p>
RSEM for diauxic shift
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/RSEM_util/run_RSEM_align_n_estimate.pl \
  --transcripts trinity_out_dir/Trinity.fasta \
  --left Sp.ds.1M.left.fq \
  --right Sp.ds.1M.right.fq \
  --seqType fq \
  --SS_lib_type RF \
  --prefix DS</code></pre>
</div></div>
</li>
<li>
<p>
RSEM for heat shock
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/RSEM_util/run_RSEM_align_n_estimate.pl \
  --transcripts trinity_out_dir/Trinity.fasta \
  --left Sp.hs.1M.left.fq \
  --right Sp.hs.1M.right.fq \
  --seqType fq \
  --SS_lib_type RF \
  --prefix HS</code></pre>
</div></div>
</li>
<li>
<p>
RSEM for plateau phase
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/RSEM_util/run_RSEM_align_n_estimate.pl \
   --transcripts trinity_out_dir/Trinity.fasta \
   --left Sp.plat.1M.left.fq \
   --right Sp.plat.1M.right.fq \
   --seqType fq \
   --SS_lib_type RF \
   --prefix PLAT</code></pre>
</div></div>
</li>
</ol></div>
</div>
<div class="sect2">
<h3 id="_differential_expression_analysis_using_edger_timing_lt_5_min">Differential expression analysis using edgeR Timing: &lt;5 min</h3>
<div class="admonitionblock">
<table><tr>
<td class="icon">
<div class="title">Note</div>
</td>
<td class="content">In this section, the genes and transcripts can be examined separately using their corresponding RSEM abundance estimates. For brevity, we pursue here only the transcripts below.</td>
</tr></table>
</div>
<div class="olist arabic"><ol class="arabic" start="19">
<li>
<p>
You will see that each of the RSEM ‘*.isoforms.results’ files has a number of columns, but we only need the one called ‘expected_count’. Create a matrix containing the counts of RNA-seq fragments per feature in a simple tab-delimited text file using the expected fragment count data produced by RSEM.
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/RSEM_util/merge_RSEM_frag_counts_single_table.pl\
    LOG.isoforms.results DS.isoforms.results HS.isoforms.results \
    PLAT.isoforms.results &gt; Sp_isoforms.counts.matrix</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>Please note that the first column of the resulting matrix is the name of the transcript. The second, third and so on are the raw counts for each of the corresponding samples. The first row contains the column headings including a label for each sample.</p></div>
<div class="olist arabic"><ol class="arabic" start="20">
<li>
<p>
Use edgeR to identify differentially expressed transcripts for each pair of samples. The following script automates
many of the tasks of running edgeR or DESeq; in this procedure, we only leverage edgeR. Use the matrix created in
Step 19 as input.
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/Analysis/DifferentialExpression/run_DE_analysis.pl \
  --matrix Sp_isoforms.counts.matrix \
  --method edgeR \
  --output edgeR_dir</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>Please note that all the edgeR results from the pairwise comparisons now exist in the ‘edgeR_dir/’ output directory, and also include the following files of interest: *.edgeR.DE_results (differentially expressed transcripts identified, including fold change and statistical significance (Table 3)) and *.edgeR.DE_results.MA_n_Volcano.pdf (MA and volcano plots from pairwise comparisons (Fig. 9)).</p></div>
<div class="olist arabic"><ol class="arabic" start="21">
<li>
<p>
To perform TMM normalization and to generate a matrix of expression values measured in FPKM, first extract the transcript length values from any one of RSEM’s *.isoform.results files:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% cut -f1,3,4 DS.isoforms.results &gt; Trinity.trans_lengths.txt</code></pre>
</div></div>
</li>
<li>
<p>
Now, perform the TMM normalization:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/Analysis/DifferentialExpression/run_TMM_normalization_write_FPKM_matrix.pl \
  --matrix Sp_isoforms.counts.matrix \
  --lengths Trinity.trans_lengths.txt</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>This command will generate the following files: ‘Sp_isoforms.counts.matrix.TMM_info.txt’, containing the effective library size for each sample after TMM normalization; and ‘Sp_isoforms.counts.matrix.TMM_normalized.FPKM’, which contains normalized transcript expression values according to the transcript and sample, measured as FPKM. This matrix file will be used for clustering expression profiles for transcripts across samples and generating heat map visualizations, as described below.</p></div>
<div class="olist arabic"><ol class="arabic" start="23">
<li>
<p>
To study expression patterns of transcripts or genes across samples, it is often useful to restrict analysis to those transcripts that are significantly differentially expressed in at least one pairwise sample comparison. Given a set of differentially expressed transcripts, extract their normalized expression values and perform hierarchical clustering to group together transcripts with similar expression patterns across samples, and to group together those samples that have similar expression profiles according to transcripts. For example, enter the ‘edgeR_dir/’ output directory and extract those transcripts that are at least fourfold differentially expressed with false discovery–corrected statistical significance of at most 0.001 by using the following commands:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% cd edgeR_dir/
% $TRINITY_HOME/Analysis/DifferentialExpression/analyze_diff_expr.pl \
  --matrix ./Sp_isoforms.counts.matrix.TMM_normalized.FPKM \
  -C 2 -P 0.001</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>Please note that the -C parameter takes the log2 (fold_change) cutoff, which in this case is log2(4) = 2. A number of files are generated, all with the prefix ‘diffExpr.P0.001_C2’ indicating the parameter choices: ‘diffExpr.P0.001_C2.matrix’ contains the subset of transcripts from the complete matrix ‘matrix.TMM_normalized.FPKM’ that were identified as differentially expressed, as defined by the specified thresholds. ‘diffExpr.P0.001_C2.matrix.heatmap.pdf’ contains a clustered heat map image showing the relationships among transcripts and samples (Fig. 10a) and a heat map of the pair-wise Spearman
correlations between samples (Fig. 10b). ‘diffExpr.P0.001_C2.matrix.R.all.RData’ is a local storage of all the data generated during this analysis, which is used further down in the PROCEDURE (Step 25) with additional analysis tools.</p></div>
<div class="olist arabic"><ol class="arabic" start="24">
<li>
<p>
Determine the number of such differentially expressed transcripts by counting the number of lines in the file by using the command:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% wc -l diffExpr.P0.001_C2.matrix</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>Subtract 1 so that you do not count the column header line as a transcript entry. Note that the ‘analyze_diff_expr.pl’ script will also directly report the number of differentially expressed transcripts identified at the given thresholds.</p></div>
<div class="olist arabic"><ol class="arabic" start="25">
<li>
<p>
Extract clusters of transcripts with common expression profiles from the earlier generated hierarchical clusters by running the script below, which uses R to cut the tree representing the hierarchically clustered transcripts based on specified criteria, such as to generate a specific number of clusters or by cutting the tree at a certain height. For example, run the following to partition transcripts by cutting the tree at 20% of its height:
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/Analysis/DifferentialExpression/define_clusters_by_cutting_tree.pl \
  --Ptree 20 -R diffExpr.P0.001_C2.matrix.R.all.Rdata</code></pre>
</div></div>
</li>
</ol></div>
<div class="paragraph"><p>The above command generates a directory (‘diffExpr.P0.001_C2.matrix.R.all.RData.clusters_fixed_P_20/’) that contains ‘subcluster_*_log2_medianCentered_fpkm.matrix’—each autodefined cluster of transcripts is provided along with expression values that are log2-transformed and median-centered—and ‘my_cluster_plots.pdf’—contains a plot of the log2-transformed, median-centered expression values for each cluster (Fig. 10c). Note that, owing to the wide dynamic range in expression values of transcripts, during this step, the expression values were first log2-transformed before plotting data points. In addition, in order to examine common expression patterns that focus on the relative expression of transcripts across multiple samples, each transcript’s expression value was subsequently centered by the median value. This operation was performed by subtracting each transcript’s median log2(FPKM) value from its log2(FPKM) value in each sample. These resulting data are referred to as log2-transformed, median-centered expression values, as generated in this step.</p></div>
<div class="olist arabic"><ol class="arabic" start="26">
<li>
<p>
(Optional) Run the script in Step 25 several times with different values of ‘--Ptree’ in order to increase or decrease the number of clusters generated.
Automating the required sections of the PROCEDURE ● TIMING 2–3 h
</p>
</li>
<li>
<p>
(Optional) If you are interested in executing the sections of the PROCEDURE without manually typing in each command, run the script below, which executes the required sections of the PROCEDURE. These sections include concatenating all samples’ reads into a single input data set, assembling the reads using Trinity, performing abundance estimation separately for each sample and running edgeR to identify differentially expressed transcripts. Run the automated procedure with the following command, including the -I (optional) parameter for an interactive experience, in which the system will pause and wait for a user response before proceeding to the next step.
</p>
<div class="literalblock">
<div class="content">
<pre><code>% $TRINITY_HOME/util/run_Trinity_edgeR_pipeline.pl --samples_file samples_n_reads_described.txt -I</code></pre>
</div></div>
</li>
</ol></div>
</div>
</div>
</div>
</div>
<div id="footnotes"><hr /></div>
<div id="footer">
<div id="footer-text">
Last updated 2013-06-21 10:02:15 EDT
</div>
</div>
</body>
</html>
