<!DOCTYPE html>
<html>
<head lang="en">
    <meta charset="UTF-8">
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    <title>Trinity Performance Metrics</title>
    <link rel="shortcut icon" href="favicon.ico" type="image/x-icon">
    <link rel="icon" href="favicon.ico" type="image/x-icon">

    <script type="text/javascript" src="//ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
    <script src="//code.highcharts.com/highcharts.js"></script>
    <script src="gray.js"></script>
    <script src="performance.js"></script>
    <link rel="stylesheet" type="text/css" href="styles.css">
    <script src="//code.highcharts.com/modules/exporting.js"></script>
    <style type="text/css">
        ${demo.css}
    </style>
</head>
<body>
<div class="navbar">
    <ul>
        <li><a href="http://trinityrnaseq.github.io/">Trinity</a></li>
        <li><a href="#">Performance</a></li>
        <li><a href="requirements.html">Requirements</a></li>
        <li><a href="benchmark.html">Benchmarks</a></li>
    </ul>
</div>
<img class="background" src="/images/TrinityCompositeLogo.png" />
<h1>Performance</h1>
<p>All runs were performed using Trinity 2.1.1 with 120G of RAM, 16 CPU's, and the --bflyCalculateCPU parameter. Each
    data set consists of approximately 50 million reads taken from the described organism.
    Tests were performed on Indiana University's Karst cluster, which consists of IBM NeXtScale nx360 M4 servers,
    each equipped with two Intel Xeon E5-2650 v2 8-core processors.
    Tests were performed on nodes with 512GB of memory.
</p>
<h2>Transcript Reconstruction Performance</h2>
<p>Since Trinity attempts to reconstruct RNA transcripts, not a genome, there is no single sequence for what Trinity
    calls a gene. A Trinity gene is a collection of related transcripts. In a de novo assembly, the transcripts are
    related by having shared kmers during the assembly stage.
</p>
<table style="width:90%">
    <tr>
        <td id="fly_genes" style="width: 30%"></td>
        <td id="yeast_genes" style="width: 30%"></td>
        <td id="mouse_genes" style="width: 30%"></td>
    </tr>
    </table>
<table style="width:90%">
    <tr>
        <td id="fly" style="width: 30%"></td>
        <td id="yeast" style="width: 30%"></td>
        <td id="mouse" style="width: 30%"></td>
    </tr>
</table>
<h2>Runtime</h2>
<p>The total amount of wall time required to produce an assembly. <span class="species">S. pombe</span> times do not include jaccard clipping.
</p>
<table style="width:90%">
    <tr>
        <td id="fly_time" style="width: 30%"></td>
        <td id="yeast_time" style="width: 30%"></td>
        <td id="mouse_time" style="width: 30%"></td>
    </tr>
</table>
<p>The many components of Trinity each have their own unique processor requirements. Additional information can
    be found on the <a href="cpu.html">Trinity Processor Usage</a> page.</p>
<h2>Memory Usage</h2>
<p>The maximum amount of memory used at a single time when producing an assembly. This number does not include
    memory used by the parallel Butterfly component, which can use as much or as little memory as is available
    on the system. (See <a href="mem.html">Trinity Memory Usage</a> for more information)</p>
</p>
<table style="width:90%">
    <tr>
        <td id="fly_mem" style="width: 30%"></td>
        <td id="yeast_mem" style="width: 30%"></td>
        <td id="mouse_mem" style="width: 30%"></td>
    </tr>
</table>
<p>The many components of Trinity each have their own unique memory requirements. Additional information can
    be found on the <a href="mem.html">Trinity Memory Usage</a> page.</p>
<h4>References</h4>
<a name="reference1"></a>
1. <cite>Bo Li, Nathanael Fillmore, Yongsheng Bai, Mike Collins, James A. Thompson, Ron Stewart, Colin N. Dewey.
    Evaluation of de novo transcriptome assemblies from RNA-Seq data.</cite>
<script type="application/javascript">
    function LightenDarkenColor(col,amt) {
        col = parseInt(col.substring(1),16);
        return '#' + ((((col & 0x0000FF) + amt) | ((((col>> 8) & 0x00FF) + amt) << 8) | (((col >> 16) + amt) << 16)).toString(16));
    }

    const cats = ['2013 Feb', '2013 Aug', '2013 Nov', '2014 Apr', '2014 Jul', 'v2.0.6', 'v2.1.1'];

    function timeFormat(ms) {
        x = ms / 1000;
        seconds = x % 60;
        x /= 60;
        minutes = x % 60;
        hours = (x-minutes)/60;
        return (hours < 10 ? '0' : '') + hours + ':' + (minutes < 10 ? '0' : '') + minutes;
    }
    function create_detonate_chart(title, target, ymin, series, color) {
        $(target).highcharts({
            chart: {
                type: 'bar', height: '325'
            },
            title: {
                text: title
            },
            subtitle: {
                text: 'Detonate Evaluation'
            },
            xAxis: {
                categories: cats,
                title: {
                    text: null
                }
            },
            yAxis: {
                min: ymin,
                title: {
                    text: 'Detonate rating (shorter is better!)'
                 },
                labels: {
                    formatter: function () {
                        return "-" + this.value.toExponential();
                    }
                }
            },
            tooltip: {
                formatter: function () {
                    return "-" + this.y.toExponential();
                }
            },

            credits: {
                enabled: false
            },
            plotOptions: { bar: { pointWidth: 5 }},
            series: [
                {
                    name: 'RSEM-EVAL Score',
                    data: series,
                    color: Highcharts.theme.colors[ color]
                }
            ]
        });
    }

    function create_gene_chart(title, target, min, gene_data, transcript_data, color) {
        $(target).highcharts({
            chart: {
                type: 'bar', height: 300
            },
            title: {
                text: title
            },
            subtitle: {
                text: 'Gene and Transcript Counts'
            },
            xAxis: {
                categories: cats
            },
            yAxis: {
                min: min,
            },
            legend: {
                reversed: true
            },
            plotOptions: { bar: { pointWidth: 5 }},
            series: [
                {
                    name: 'Transcript Count',
                    data: transcript_data,
                    color: LightenDarkenColor(Highcharts.theme.colors[color], -10)
                },
                {
                    name: '"Gene" Count',
                    data: gene_data,
                    color: Highcharts.theme.colors[ color]
                }
            ]
        });
    }

    function create_time_chart(title, target, data, color) {
        $(target).highcharts({
            chart: {
                type: 'bar', height: 300
            },
            title: {
                text: title
            },
            subtitle: {
                text: 'Wall Time'
            },
            xAxis: {
                categories: cats
            },
            yAxis: {
                title: {
                    text: 'Total Time'
                },
                type: 'datetime',
                labels: {
                    formatter: function () {
                        return timeFormat(this.value);
                    }}
            },
            tooltip: {
                formatter: function() {
                    return timeFormat(this.y);
                }
            },
            legend: {
                reversed: true
            },
            plotOptions: { bar: { pointWidth: 5 }},
            series: [
                {
                    name: 'Total Time',
                    data: data,
                    color: Highcharts.theme.colors[ color]
                }
            ]
        });
    }

    function create_memory_chart(title, target, data, color) {
        $(target).highcharts({
            chart: {
                type: 'bar', height: 300
            },
            title: {
                text: title
            },
            subtitle: {
                text: 'Memory'
            },
            xAxis: {
                categories: cats
            },
            yAxis: {
                title: {
                    text: 'Memory'
                }
            },
            legend: {
                reversed: true
            },
            plotOptions: { bar: { pointWidth: 5 }},
            series: [
                {
                    name: 'Memory',
                    data: data,
                    color: Highcharts.theme.colors[ color]
                }
            ]
        });
    }

    function create_charts(data) {
        var title = data['title'];
        var prefix = data['prefix'];
        var color = data['color'];
        create_detonate_chart(title, prefix, data['detonate_min'], data['detonate'], color);
        create_gene_chart(title, prefix + '_genes', data['gene_count_min'], data['gene_count'], data['total_transcripts'], color);
        create_time_chart(title, prefix + '_time', data['wall_time'], color);
        create_memory_chart(title, prefix + '_mem', data['max_mem'], color);
    }
    $(function () {
        var droso = { detonate : [2505042892, 2506637187, 2510591687, 2476429902, 2487879464, 2461149402, 2457674377],
            detonate_min : 2400000000,
            gene_count : [44481, 44465,	44494, 45172, 45165, 50156],
            gene_count_min : 6000,
            total_transcripts : [55569, 55550, 58950, 51622, 53646, 55066],
            gene_count_min : 6000,
            wall_time : [12223000.00, 11717000.00, 8740000.00, 8970000.00, 14992000.00, 20004000.00],
            max_mem : [137915011072, 137905659904, 137906634752, 137915211776, 133894221824, 133892780032],
            prefix: '#fly',
            title : 'D. melanogaster',
            color : 2};

        var mouse = { detonate : [3740144504, 3727712340, 3738177439, 3698945681, 3690981994, 3652310504, 3639739766],
            detonate_min : 3500000000,
            gene_count : [55420,55496, 55580, 56749, 56992, 60618, 69390],
            gene_count_min : 7000,
            total_transcripts : [80703, 80825, 90847, 67311, 72918, 71536, 80496],
            wall_time : [17044000.00, 16076000.00, 12769000.00, 13740000.00, 24187000.00, 23370000.00, 16453000.00],
            max_mem : [129778188288, 129768828928, 129769803776, 129160675328, 129160695808, 129153875968, 129153675264],
            prefix: '#mouse',
            title : 'M. muscula',
            color : 1};

        var schizo = { detonate : [2124269914, 2114808563, 2110474455, 2096633690, 2136157533, 2076441834, 2071793893],
            detonate_min : 2000000000,
            gene_count : [11550, 11552, 11559, 11651, 11651, 13519, 14533],
            gene_count_min : 3500,
            total_transcripts : [13767, 13778, 14158, 13328, 13328, 14504, 15498],
            wall_time : [7380000.00, 7022000.00, 6123000.00, 6269000.00, 11245000.00, 9585000.00, 7322000.00],
            max_mem : [129778192384, 129768816640, 129769799680, 129168748544, 129160683520, 129153863680, 129153675264],
            prefix: '#yeast',
            title : 'S. pombe',
            color : 0};

        create_charts(droso);
        create_charts(mouse);
        create_charts(schizo);

    });

</script>
</body>
</html>