<!DOCTYPE html>
<html>
<head lang="en">
    <meta charset="UTF-8">
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    <title>Trinity Performance Metrics</title>
    <link rel="shortcut icon" href="favicon.ico" type="image/x-icon">
    <link rel="icon" href="favicon.ico" type="image/x-icon">

    <script type="text/javascript" src="//ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
    <script src="//code.highcharts.com/highcharts.js"></script>
    <script src="gray.js"></script>
    <script src="performance.js"></script>
    <link rel="stylesheet" type="text/css" href="styles.css">
    <script src="//code.highcharts.com/modules/exporting.js"></script>
    <style type="text/css">
        ${demo.css}
    </style>
</head>

<body>
<div class="navbar">
    <ul>
        <li><a href="http://trinityrnaseq.sourceforge.net/">Trinity</a></li>
        <li><a href="index.html">Performance</a></li>
        <li><a href="#">Requirements</a></li>
        <li><a href="benchmark.html">Benchmarks</a></li>
    </ul>
</div>
<img class="background" src="/images/TrinityCompositeLogo.png" />
<h1>Performance</h1>
<h2>Hardware and Configuration Requirements</h2>
<p>The Inchworm and Chrysalis steps can be memory intensive.  A basic recommendation is to have
    1G of RAM per 1M pairs of Illumina reads. Simpler transcriptomes (lower eukaryotes) require
    less memory than more complex transcriptomes such as from vertebrates.</p>
<p>If you are able to run the entire Trinity process on a single high-memory multi-core server,
    indicate the number of butterfly processes to run in parallel by the --CPU parameter.</p>
<p>Our experience is that the entire process can require ~1/2 hour to one hour per million pairs
    of reads in the current implementation (see <a href="http://trinityrnaseq.sourceforge.net/trinity_faq.html">FAQ</a>).  We're striving
    to improve upon both memory and time requirements.</p>
<p>If you do not have access to a high memory machine (typically having 256G or 512G of RAM),
    consider running Trinity on one of the externally available resources. If you are limited to the amount of
    time available for executing Trinity (due to artificially imposed limits on a shared computing resource), you
    can aim to run Trinity in separate stages, where subsequent stages resume from the previous ones.  To do so,
    include the following options for each of the stages:</p></div>
<ol>
    <li>Generate the kmer-catalog using jellyfish: <em>--no_run_inchworm</em></li>
    <li>Run Inchworm to generate draft contigs:  <em>--no_run_chrysalis</em></li>
    <li>Chrysalis clustering of inchworm contigs and mapping reads: <em>--no_run_quantifygraph</em></li>
    <li>Chrysalis deBruijn graph construction: <em>--no_run_butterfly</em></li>
    <li>Run butterfly, generate final Trinity.fasta file:  (exclude <em>--no_</em> options)</li>
</ol>

</body>
</html>