<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
    "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta http-equiv="Content-Type" content="application/xhtml+xml; charset=UTF-8" />
<meta name="generator" content="AsciiDoc 8.6.8" />
<title>Identification and Analysis of Differentially Expressed Trinity Genes and Transcripts</title>
<style type="text/css">
/* Shared CSS for AsciiDoc xhtml11 and html5 backends */

/* Default font. */
body {
  font-family: Georgia,serif;
}

/* Title font. */
h1, h2, h3, h4, h5, h6,
div.title, caption.title,
thead, p.table.header,
#toctitle,
#author, #revnumber, #revdate, #revremark,
#footer {
  font-family: Arial,Helvetica,sans-serif;
}

body {
  margin: 1em 5% 1em 5%;
}

a {
  color: blue;
  text-decoration: underline;
}
a:visited {
  color: fuchsia;
}

em {
  font-style: italic;
  color: navy;
}

strong {
  font-weight: bold;
  color: #083194;
}

h1, h2, h3, h4, h5, h6 {
  color: #527bbd;
  margin-top: 1.2em;
  margin-bottom: 0.5em;
  line-height: 1.3;
}

h1, h2, h3 {
  border-bottom: 2px solid silver;
}
h2 {
  padding-top: 0.5em;
}
h3 {
  float: left;
}
h3 + * {
  clear: left;
}
h5 {
  font-size: 1.0em;
}

div.sectionbody {
  margin-left: 0;
}

hr {
  border: 1px solid silver;
}

p {
  margin-top: 0.5em;
  margin-bottom: 0.5em;
}

ul, ol, li > p {
  margin-top: 0;
}
ul > li     { color: #aaa; }
ul > li > * { color: black; }

.monospaced, code, pre {
  font-family: "Courier New", Courier, monospace;
  font-size: inherit;
  color: navy;
  padding: 0;
  margin: 0;
}


#author {
  color: #527bbd;
  font-weight: bold;
  font-size: 1.1em;
}
#email {
}
#revnumber, #revdate, #revremark {
}

#footer {
  font-size: small;
  border-top: 2px solid silver;
  padding-top: 0.5em;
  margin-top: 4.0em;
}
#footer-text {
  float: left;
  padding-bottom: 0.5em;
}
#footer-badges {
  float: right;
  padding-bottom: 0.5em;
}

#preamble {
  margin-top: 1.5em;
  margin-bottom: 1.5em;
}
div.imageblock, div.exampleblock, div.verseblock,
div.quoteblock, div.literalblock, div.listingblock, div.sidebarblock,
div.admonitionblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
div.admonitionblock {
  margin-top: 2.0em;
  margin-bottom: 2.0em;
  margin-right: 10%;
  color: #606060;
}

div.content { /* Block element content. */
  padding: 0;
}

/* Block element titles. */
div.title, caption.title {
  color: #527bbd;
  font-weight: bold;
  text-align: left;
  margin-top: 1.0em;
  margin-bottom: 0.5em;
}
div.title + * {
  margin-top: 0;
}

td div.title:first-child {
  margin-top: 0.0em;
}
div.content div.title:first-child {
  margin-top: 0.0em;
}
div.content + div.title {
  margin-top: 0.0em;
}

div.sidebarblock > div.content {
  background: #ffffee;
  border: 1px solid #dddddd;
  border-left: 4px solid #f0f0f0;
  padding: 0.5em;
}

div.listingblock > div.content {
  border: 1px solid #dddddd;
  border-left: 5px solid #f0f0f0;
  background: #f8f8f8;
  padding: 0.5em;
}

div.quoteblock, div.verseblock {
  padding-left: 1.0em;
  margin-left: 1.0em;
  margin-right: 10%;
  border-left: 5px solid #f0f0f0;
  color: #888;
}

div.quoteblock > div.attribution {
  padding-top: 0.5em;
  text-align: right;
}

div.verseblock > pre.content {
  font-family: inherit;
  font-size: inherit;
}
div.verseblock > div.attribution {
  padding-top: 0.75em;
  text-align: left;
}
/* DEPRECATED: Pre version 8.2.7 verse style literal block. */
div.verseblock + div.attribution {
  text-align: left;
}

div.admonitionblock .icon {
  vertical-align: top;
  font-size: 1.1em;
  font-weight: bold;
  text-decoration: underline;
  color: #527bbd;
  padding-right: 0.5em;
}
div.admonitionblock td.content {
  padding-left: 0.5em;
  border-left: 3px solid #dddddd;
}

div.exampleblock > div.content {
  border-left: 3px solid #dddddd;
  padding-left: 0.5em;
}

div.imageblock div.content { padding-left: 0; }
span.image img { border-style: none; }
a.image:visited { color: white; }

dl {
  margin-top: 0.8em;
  margin-bottom: 0.8em;
}
dt {
  margin-top: 0.5em;
  margin-bottom: 0;
  font-style: normal;
  color: navy;
}
dd > *:first-child {
  margin-top: 0.1em;
}

ul, ol {
    list-style-position: outside;
}
ol.arabic {
  list-style-type: decimal;
}
ol.loweralpha {
  list-style-type: lower-alpha;
}
ol.upperalpha {
  list-style-type: upper-alpha;
}
ol.lowerroman {
  list-style-type: lower-roman;
}
ol.upperroman {
  list-style-type: upper-roman;
}

div.compact ul, div.compact ol,
div.compact p, div.compact p,
div.compact div, div.compact div {
  margin-top: 0.1em;
  margin-bottom: 0.1em;
}

tfoot {
  font-weight: bold;
}
td > div.verse {
  white-space: pre;
}

div.hdlist {
  margin-top: 0.8em;
  margin-bottom: 0.8em;
}
div.hdlist tr {
  padding-bottom: 15px;
}
dt.hdlist1.strong, td.hdlist1.strong {
  font-weight: bold;
}
td.hdlist1 {
  vertical-align: top;
  font-style: normal;
  padding-right: 0.8em;
  color: navy;
}
td.hdlist2 {
  vertical-align: top;
}
div.hdlist.compact tr {
  margin: 0;
  padding-bottom: 0;
}

.comment {
  background: yellow;
}

.footnote, .footnoteref {
  font-size: 0.8em;
}

span.footnote, span.footnoteref {
  vertical-align: super;
}

#footnotes {
  margin: 20px 0 20px 0;
  padding: 7px 0 0 0;
}

#footnotes div.footnote {
  margin: 0 0 5px 0;
}

#footnotes hr {
  border: none;
  border-top: 1px solid silver;
  height: 1px;
  text-align: left;
  margin-left: 0;
  width: 20%;
  min-width: 100px;
}

div.colist td {
  padding-right: 0.5em;
  padding-bottom: 0.3em;
  vertical-align: top;
}
div.colist td img {
  margin-top: 0.3em;
}

@media print {
  #footer-badges { display: none; }
}

#toc {
  margin-bottom: 2.5em;
}

#toctitle {
  color: #527bbd;
  font-size: 1.1em;
  font-weight: bold;
  margin-top: 1.0em;
  margin-bottom: 0.1em;
}

div.toclevel0, div.toclevel1, div.toclevel2, div.toclevel3, div.toclevel4 {
  margin-top: 0;
  margin-bottom: 0;
}
div.toclevel2 {
  margin-left: 2em;
  font-size: 0.9em;
}
div.toclevel3 {
  margin-left: 4em;
  font-size: 0.9em;
}
div.toclevel4 {
  margin-left: 6em;
  font-size: 0.9em;
}

span.aqua { color: aqua; }
span.black { color: black; }
span.blue { color: blue; }
span.fuchsia { color: fuchsia; }
span.gray { color: gray; }
span.green { color: green; }
span.lime { color: lime; }
span.maroon { color: maroon; }
span.navy { color: navy; }
span.olive { color: olive; }
span.purple { color: purple; }
span.red { color: red; }
span.silver { color: silver; }
span.teal { color: teal; }
span.white { color: white; }
span.yellow { color: yellow; }

span.aqua-background { background: aqua; }
span.black-background { background: black; }
span.blue-background { background: blue; }
span.fuchsia-background { background: fuchsia; }
span.gray-background { background: gray; }
span.green-background { background: green; }
span.lime-background { background: lime; }
span.maroon-background { background: maroon; }
span.navy-background { background: navy; }
span.olive-background { background: olive; }
span.purple-background { background: purple; }
span.red-background { background: red; }
span.silver-background { background: silver; }
span.teal-background { background: teal; }
span.white-background { background: white; }
span.yellow-background { background: yellow; }

span.big { font-size: 2em; }
span.small { font-size: 0.6em; }

span.underline { text-decoration: underline; }
span.overline { text-decoration: overline; }
span.line-through { text-decoration: line-through; }

div.unbreakable { page-break-inside: avoid; }


/*
 * xhtml11 specific
 *
 * */

div.tableblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
div.tableblock > table {
  border: 3px solid #527bbd;
}
thead, p.table.header {
  font-weight: bold;
  color: #527bbd;
}
p.table {
  margin-top: 0;
}
/* Because the table frame attribute is overriden by CSS in most browsers. */
div.tableblock > table[frame="void"] {
  border-style: none;
}
div.tableblock > table[frame="hsides"] {
  border-left-style: none;
  border-right-style: none;
}
div.tableblock > table[frame="vsides"] {
  border-top-style: none;
  border-bottom-style: none;
}


/*
 * html5 specific
 *
 * */

table.tableblock {
  margin-top: 1.0em;
  margin-bottom: 1.5em;
}
thead, p.tableblock.header {
  font-weight: bold;
  color: #527bbd;
}
p.tableblock {
  margin-top: 0;
}
table.tableblock {
  border-width: 3px;
  border-spacing: 0px;
  border-style: solid;
  border-color: #527bbd;
  border-collapse: collapse;
}
th.tableblock, td.tableblock {
  border-width: 1px;
  padding: 4px;
  border-style: solid;
  border-color: #527bbd;
}

table.tableblock.frame-topbot {
  border-left-style: hidden;
  border-right-style: hidden;
}
table.tableblock.frame-sides {
  border-top-style: hidden;
  border-bottom-style: hidden;
}
table.tableblock.frame-none {
  border-style: hidden;
}

th.tableblock.halign-left, td.tableblock.halign-left {
  text-align: left;
}
th.tableblock.halign-center, td.tableblock.halign-center {
  text-align: center;
}
th.tableblock.halign-right, td.tableblock.halign-right {
  text-align: right;
}

th.tableblock.valign-top, td.tableblock.valign-top {
  vertical-align: top;
}
th.tableblock.valign-middle, td.tableblock.valign-middle {
  vertical-align: middle;
}
th.tableblock.valign-bottom, td.tableblock.valign-bottom {
  vertical-align: bottom;
}


/*
 * manpage specific
 *
 * */

body.manpage h1 {
  padding-top: 0.5em;
  padding-bottom: 0.5em;
  border-top: 2px solid silver;
  border-bottom: 2px solid silver;
}
body.manpage h2 {
  border-style: none;
}
body.manpage div.sectionbody {
  margin-left: 3em;
}

@media print {
  body.manpage div#toc { display: none; }
}


</style>
<script type="text/javascript">
/*<![CDATA[*/
var asciidoc = {  // Namespace.

/////////////////////////////////////////////////////////////////////
// Table Of Contents generator
/////////////////////////////////////////////////////////////////////

/* Author: Mihai Bazon, September 2002
 * http://students.infoiasi.ro/~mishoo
 *
 * Table Of Content generator
 * Version: 0.4
 *
 * Feel free to use this script under the terms of the GNU General Public
 * License, as long as you do not remove or alter this notice.
 */

 /* modified by Troy D. Hanson, September 2006. License: GPL */
 /* modified by Stuart Rackham, 2006, 2009. License: GPL */

// toclevels = 1..4.
toc: function (toclevels) {

  function getText(el) {
    var text = "";
    for (var i = el.firstChild; i != null; i = i.nextSibling) {
      if (i.nodeType == 3 /* Node.TEXT_NODE */) // IE doesn't speak constants.
        text += i.data;
      else if (i.firstChild != null)
        text += getText(i);
    }
    return text;
  }

  function TocEntry(el, text, toclevel) {
    this.element = el;
    this.text = text;
    this.toclevel = toclevel;
  }

  function tocEntries(el, toclevels) {
    var result = new Array;
    var re = new RegExp('[hH]([1-'+(toclevels+1)+'])');
    // Function that scans the DOM tree for header elements (the DOM2
    // nodeIterator API would be a better technique but not supported by all
    // browsers).
    var iterate = function (el) {
      for (var i = el.firstChild; i != null; i = i.nextSibling) {
        if (i.nodeType == 1 /* Node.ELEMENT_NODE */) {
          var mo = re.exec(i.tagName);
          if (mo && (i.getAttribute("class") || i.getAttribute("className")) != "float") {
            result[result.length] = new TocEntry(i, getText(i), mo[1]-1);
          }
          iterate(i);
        }
      }
    }
    iterate(el);
    return result;
  }

  var toc = document.getElementById("toc");
  if (!toc) {
    return;
  }

  // Delete existing TOC entries in case we're reloading the TOC.
  var tocEntriesToRemove = [];
  var i;
  for (i = 0; i < toc.childNodes.length; i++) {
    var entry = toc.childNodes[i];
    if (entry.nodeName.toLowerCase() == 'div'
     && entry.getAttribute("class")
     && entry.getAttribute("class").match(/^toclevel/))
      tocEntriesToRemove.push(entry);
  }
  for (i = 0; i < tocEntriesToRemove.length; i++) {
    toc.removeChild(tocEntriesToRemove[i]);
  }

  // Rebuild TOC entries.
  var entries = tocEntries(document.getElementById("content"), toclevels);
  for (var i = 0; i < entries.length; ++i) {
    var entry = entries[i];
    if (entry.element.id == "")
      entry.element.id = "_toc_" + i;
    var a = document.createElement("a");
    a.href = "#" + entry.element.id;
    a.appendChild(document.createTextNode(entry.text));
    var div = document.createElement("div");
    div.appendChild(a);
    div.className = "toclevel" + entry.toclevel;
    toc.appendChild(div);
  }
  if (entries.length == 0)
    toc.parentNode.removeChild(toc);
},


/////////////////////////////////////////////////////////////////////
// Footnotes generator
/////////////////////////////////////////////////////////////////////

/* Based on footnote generation code from:
 * http://www.brandspankingnew.net/archive/2005/07/format_footnote.html
 */

footnotes: function () {
  // Delete existing footnote entries in case we're reloading the footnodes.
  var i;
  var noteholder = document.getElementById("footnotes");
  if (!noteholder) {
    return;
  }
  var entriesToRemove = [];
  for (i = 0; i < noteholder.childNodes.length; i++) {
    var entry = noteholder.childNodes[i];
    if (entry.nodeName.toLowerCase() == 'div' && entry.getAttribute("class") == "footnote")
      entriesToRemove.push(entry);
  }
  for (i = 0; i < entriesToRemove.length; i++) {
    noteholder.removeChild(entriesToRemove[i]);
  }

  // Rebuild footnote entries.
  var cont = document.getElementById("content");
  var spans = cont.getElementsByTagName("span");
  var refs = {};
  var n = 0;
  for (i=0; i<spans.length; i++) {
    if (spans[i].className == "footnote") {
      n++;
      var note = spans[i].getAttribute("data-note");
      if (!note) {
        // Use [\s\S] in place of . so multi-line matches work.
        // Because JavaScript has no s (dotall) regex flag.
        note = spans[i].innerHTML.match(/\s*\[([\s\S]*)]\s*/)[1];
        spans[i].innerHTML =
          "[<a id='_footnoteref_" + n + "' href='#_footnote_" + n +
          "' title='View footnote' class='footnote'>" + n + "</a>]";
        spans[i].setAttribute("data-note", note);
      }
      noteholder.innerHTML +=
        "<div class='footnote' id='_footnote_" + n + "'>" +
        "<a href='#_footnoteref_" + n + "' title='Return to text'>" +
        n + "</a>. " + note + "</div>";
      var id =spans[i].getAttribute("id");
      if (id != null) refs["#"+id] = n;
    }
  }
  if (n == 0)
    noteholder.parentNode.removeChild(noteholder);
  else {
    // Process footnoterefs.
    for (i=0; i<spans.length; i++) {
      if (spans[i].className == "footnoteref") {
        var href = spans[i].getElementsByTagName("a")[0].getAttribute("href");
        href = href.match(/#.*/)[0];  // Because IE return full URL.
        n = refs[href];
        spans[i].innerHTML =
          "[<a href='#_footnote_" + n +
          "' title='View footnote' class='footnote'>" + n + "</a>]";
      }
    }
  }
},

install: function(toclevels) {
  var timerId;

  function reinstall() {
    asciidoc.footnotes();
    if (toclevels) {
      asciidoc.toc(toclevels);
    }
  }

  function reinstallAndRemoveTimer() {
    clearInterval(timerId);
    reinstall();
  }

  timerId = setInterval(reinstall, 500);
  if (document.addEventListener)
    document.addEventListener("DOMContentLoaded", reinstallAndRemoveTimer, false);
  else
    window.onload = reinstallAndRemoveTimer;
}

}
asciidoc.install();
/*]]>*/
</script>
</head>
<body class="article">
<div id="header">
<h1>Identification and Analysis of Differentially Expressed Trinity Genes and Transcripts</h1>
</div>
<div id="content">
<div id="preamble">
<div class="sectionbody">
<div class="paragraph"><p><a href="../index.html">Trinity-home</a></p></div>
<div class="paragraph"><p>Our current system for identifying differentially expressed transcripts relies on using the EdgeR Bioconductor package. We have a protocol and scripts described below for identifying differentially expressed transcripts and clustering transcripts according to expression profiles. This process is somewhat interactive, and described are automated approaches as well as manual approaches to refining gene clusters and examining their corresponding expression patterns.</p></div>
<div class="paragraph"><p>We recommend generating a single Trinity assembly based on combining all reads across all samples as inputs.  Then, reads are separately aligned back to the single Trinity assembly for downstream analyses of differential expression, according to our <a href="abundance_estimation.html">abundance estimation protocol</a>.   If you decide to assemble each sample separately, then you&#8217;ll likely have difficulty comparing the results across the different samples due to differences in assembled transcript lengths and contiguity.</p></div>
<div class="admonitionblock">
<table><tr>
<td class="icon">
<div class="title">Note</div>
</td>
<td class="content">If you have biological replicates, align each replicate set of reads and estimate abundance values for the Trinity contigs independently.</td>
</tr></table>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_table_of_contents">Table of Contents</h2>
<div class="sectionbody">
<div class="ulist"><ul>
<li>
<p>
<a href="#ExprValueMatrix">Generating Expression Value Matrices</a>
</p>
</li>
<li>
<p>
<a href="#DE_identify">Identifying Differentially Expressed Transcripts</a>
</p>
<div class="ulist"><ul>
<li>
<p>
<a href="#no_bio_replicates">Identifying DE features: No biological replicates</a>
</p>
</li>
<li>
<p>
<a href="#with_bio_replicates">Identifying DE features: With biological replicates</a>
</p>
</li>
</ul></div>
</li>
<li>
<p>
<a href="#DE_analysis">Analyzing the Differentially Expressed Transcripts</a>
</p>
<div class="ulist"><ul>
<li>
<p>
<a href="#extract_and_cluster">Extracting and clustering differentially expressed transcripts</a>
</p>
</li>
<li>
<p>
<a href="#expr_clustering">Automatically partitioning genes into expression clusters</a>
</p>
</li>
<li>
<p>
<a href="#manually_define_clusters">Manually defining gene clusters</a>
</p>
</li>
</ul></div>
</li>
</ul></div>
</div>
</div>
<div class="sect1">
<h2 id="ExprValueMatrix">Generating Expression Value Matrices</h2>
<div class="sectionbody">
<div class="paragraph"><p>First, join the <a href="abundance_estimation.html">RSEM-estimated abundance values</a> for each of your samples by running the following, given transcript or gene-level abundance estimates from RSEM or eXpress:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/util/abundance_estimates_to_matrix.pl</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>############################################################
#
# Usage:  $TRINITY_HOME/util/abundance_estimates_to_matrix.pl --est_method &lt;method&gt;  sample1.results sample2.results ...
# Required:
#
#  --est_method &lt;string&gt;           RSEM|eXpress  (needs to know what format to expect)
#
#
# Options:
#
#  --cross_sample_fpkm_norm &lt;string&gt;    TMM|UpperQuartile|none   (default: TMM)
#
#  --name_sample_by_basedir             name sample column by dirname instead of filename
#
#  --out_prefix &lt;string&gt;                default: 'matrix'
#
############################################################</code></pre>
</div></div>
<div class="paragraph"><p>For example, given a set of RSEM <em>isoform.results</em> files for a set of samples (and naming them according to sample), the abundance matrix would be generated like so:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/util/abundance_estimates_to_matrix.pl --est_method RSEM  --out_prefix Trinity_trans sampleA.RSEM.isoform.results \
                    sampleB.RSEM.isoform.results sampleC.isoform.results sampleD.isoform.results ...</code></pre>
</div></div>
<div class="paragraph"><p>which will generate the following files:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>Trinity_trans.counts.matrix  : matrix of fragment raw counts
Trinity_trans.TMM.fpkm.matrix : TMM-normalized FPKM expression values</code></pre>
</div></div>
<div class="paragraph"><p>Do the same for the RSEM <em>gene</em> files (where counts are aggregated per Trinity <em>gene</em>) like so:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/util/abundance_estimates_to_matrix.pl --est_method RSEM  --out_prefix Trinity_genes sampleA.RSEM.gene.results \
                    sampleB.RSEM.gene.results sampleC.gene.results sampleD.gene.results ...</code></pre>
</div></div>
<div class="paragraph"><p>and you&#8217;ll find similarly named files for the counts matrix and the TMM-normalized FPKM values, with the <em>Trinity_genes</em> prefix.</p></div>
<div class="paragraph"><p>The <em>counts</em> matrix is used for differential expression analysis in Bioconductor (eg. edgeR).  The TMM-normalized FPKM matrix is used for generating heatmaps and exploring patterns of gene expression.  The TMM-normalization is performed using code in edgeR as described in <a href="http://genomebiology.com/2010/11/3/R25">Robinson and Oshlack, Genome Biology, 2010</a>, and applied to scale the FPKM values provided by the abundance estimation software (RSEM or eXpress) across all samples.</p></div>
<div class="paragraph"><p>Each downstream analysis tool will indicate which matrix type (counts or FPKM) is required.</p></div>
<div class="admonitionblock">
<table><tr>
<td class="icon">
<div class="title">Note</div>
</td>
<td class="content">If you decide to edit the column headers (sample names), be sure that they are absolutely consistent across both matrix types.  It&#8217;s easier to rename the input file used for building the matrix, since that filename becomes the corresponding column header.</td>
</tr></table>
</div>
</div>
</div>
<div class="sect1">
<h2 id="DE_identify">Identifying Differentially Expressed Transcripts</h2>
<div class="sectionbody">
<div class="paragraph"><p>Trinity currently supports the use of Bioconductor tools (edgeR and DESeq) for differential expression analysis.  To perform the steps below, you must have the <a href="http://r-project.org">R</a> software installed along with certain libraries and packages shown below.  To install Bioconductor and the required packages, run the following from an R prompt:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>source("http://bioconductor.org/biocLite.R")
biocLite('edgeR')
biocLite('DESeq')
biocLite('ctc')
biocLite('Biobase')
install.packages('gplots’)
install.packages(‘ape’)</code></pre>
</div></div>
<div class="paragraph"><p>Differentially expressed transcripts or genes are identified by running the script below, which will perform pairwise comparisons among each of your sample types. If you have biological replicates for each sample, you should indicate this as well (described further below).  To analyze transcripts, use the <em>transcripts.counts.matrix</em> file. To perform an analysis at the <em>gene</em> level, use the <em>genes.counts.matrix</em>. Again, Trinity Components are used as a proxy for <em>gene</em> level studies.</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/Analysis/DifferentialExpression/run_DE_analysis.pl</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>#################################################################################################
#
#  Required:
#
#  --matrix &lt;string&gt;               matrix of raw read counts (not normalized!)
#
#  --method &lt;string&gt;               edgeR|DESeq   (DESeq only supported here w/ bio replicates)
#
#
#  Optional:
#
#  --samples_file &lt;string&gt;         tab-delimited text file indicating biological replicate relationships.
#                                   ex.
#                                        cond_A    cond_A_rep1
#                                        cond_A    cond_A_rep2
#                                        cond_B    cond_B_rep1
#                                        cond_B    cond_B_rep2
#
#
#  General options:
#
#  --min_rowSum_counts &lt;int&gt;       default: 10  (only those rows of matrix meeting requirement will be tested)
#
#  --output|o                      aname of directory to place outputs (default: $method.$pid.dir)
#
###############################################################################################
#
#  ## EdgeR-related parameters
#  ## (no biological replicates)
#
#  --dispersion &lt;float&gt;            edgeR dispersion value (default: 0.1)   set to 0 for poisson (sometimes breaks...)
#
#  http://www.bioconductor.org/packages/release/bioc/html/edgeR.html
#
###############################################################################################
#
#  ## DE-Seq related parameters
#
#  --DESEQ_method &lt;string&gt;         "pooled", "pooled-CR", "per-condition", "blind"
#  --DESEQ_sharingMode &lt;string&gt;    "maximum", "fit-only", "gene-est-only"
#  --DESEQ_fitType &lt;string&gt;        fitType = c("parametric", "local")
#
#  ## (no biological replicates)
#        note: FIXED as: method=blind, sharingMode=fit-only
#
#  http://www.bioconductor.org/packages/release/bioc/html/DESeq.html
#
################################################################################################</code></pre>
</div></div>
<div class="admonitionblock">
<table><tr>
<td class="icon">
<div class="title">Note</div>
</td>
<td class="content">Based on our experiences with differential expression analysis and Trinity assemblies, we currently endorse edgeR as our method of choice.  DESeq is known to perform quite well when many biological replicates are available for each sample type. We aim to support additional methods in future versions of Trinity.</td>
</tr></table>
</div>
<div class="sect2">
<h3 id="no_bio_replicates">Identifying DE features: No biological replicates</h3>
<div class="paragraph"><p>Using edgeR <strong>without</strong> replicates:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/Analysis/DifferentialExpression/run_DE_analysis.pl --matrix counts.matrix --method edgeR</code></pre>
</div></div>
</div>
<div class="sect2">
<h3 id="with_bio_replicates">Identifying DE features: With biological replicates (PREFERRED)</h3>
<div class="paragraph"><p>Be sure to have a <em>samples_described.txt</em> file that describes the relationship between samples and replicates.  For example:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>conditionA   condA-rep1
conditionA   condA-rep2</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>conditionB   condB-rep1
conditionB   condB-rep2</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>conditionC   condC-rep1
conditionC   condC-rep2</code></pre>
</div></div>
<div class="paragraph"><p>where condA-rep1, condA-rep2, condB-rep1, etc&#8230;, are all column names in the <em>counts.matrix</em> generated earlier (see top of page). Your sample names that group the replicates are user-defined here.</p></div>
<div class="paragraph"><p>Using edgeR <strong>with</strong> replicates:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/Analysis/DifferentialExpression/run_DE_analysis.pl --matrix SP2.rnaseq.counts.matrix --method edgeR --samples_file samples_described.txt</code></pre>
</div></div>
<div class="admonitionblock">
<table><tr>
<td class="icon">
<div class="title">Note</div>
</td>
<td class="content">A full example of the edgeR pipeline involving combining reads from multiple samples, assembling them using Trinity, separately aligning reads back to the trintiy assemblies, abundance estimation using RSEM, and differential expression analysis using edgeR is provided at: $TRINITY_HOME/sample_data/test_full_edgeR_pipeline</td>
</tr></table>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="DE_analysis">Analyzing The Differentially Expressed Transcripts</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="extract_and_cluster">Extracting and clustering differentially expressed transcripts</h3>
<div class="paragraph"><p>An initial step in analyzing differential expression is to extract those transcripts that are most differentially expressed (most significant P-values and fold-changes) and to cluster the transcripts according to their patterns of differential expression across the samples. To do this, you can run the following from within the edgeR output directory, bu running the following script:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/Analysis/DifferentialExpression/analyze_diff_expr.pl</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>####################################################################################
#
# Required:
#
#  --matrix    matrix.normalized.FPKM
#
# Optional:
#
#  -P          p-value cutoff for FDR  (default: 0.001)
#
#  -C          min abs(log2(a/b)) fold change (default: 2  (meaning 2^(2) or 4-fold).
#
#  --output    prefix for output file (default: "diffExpr.P${Pvalue}_C${C})
#
#  Clustering methods:
#
#  --gene_dist &lt;string&gt;        euclidean, pearson, spearman,   (default: euclidean)
#                              maximum, manhattan, canberra, binary, minkowski
#
#  --gene_clust &lt;string&gt;       ward, single, complete, average, mcquitty, median, centroid (default: complete)
#
#
#
####################################################################################</code></pre>
</div></div>
<div class="paragraph"><p>For example:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/Analysis/DifferentialExpression/analyze_diff_expr.pl --matrix matrix.TMM_normalized.FPKM -P 1e-3 -C 2</code></pre>
</div></div>
<div class="paragraph"><p></p></div>
<div class="paragraph"><p><span class="image">
<img src="../images/diff_expr/clustered_heatmap.png" alt="heatmap" />
</span></p></div>
<div class="paragraph"><p>The above is mostly just a visual reference. To more seriously study and define your gene clusters, you will need to interact with the data as described below. The clusters and all required data for interrogating and defining clusters is all saved with an R-session, locally with the file <em>all.RData</em>. This will be leveraged as described below.</p></div>
</div>
<div class="sect2">
<h3 id="expr_clustering">Automatically Partitioning Genes into Expression Clusters</h3>
<div class="paragraph"><p>Run the script below to automatically split the data set into a sets of transcripts with related expression patterns by partitioning the hierarchically clustered transcript tree.</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/Analysis/DifferentialExpression/define_clusters_by_cutting_tree.pl</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>###################################################################################
#
# -K &lt;int&gt;          define K clusters via k-means algorithm
#
#  or, cut the hierarchical tree:
#
# --Ktree &lt;int&gt;     cut tree into K clusters
#
# --Ptree &lt;float&gt;   cut tree based on this percent of max(height) of tree
#
# -R &lt;string&gt;  the filename for the store RData (file.all.RData)
#
###################################################################################</code></pre>
</div></div>
<div class="paragraph"><p>For example:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/Analysis/DifferentialExpression/define_clusters_by_cutting_tree.pl --Ptree $percent_tree_height</code></pre>
</div></div>
<div class="paragraph"><p>A directory will be created called: <em>diffExpr.P0.001_C2.matrix.R.all.RData.clusters_fixed_P_20/</em> and contain the expression matrix for each of the clusters (log2-transformed, median centered).</p></div>
<div class="paragraph"><p>To plot the mean-centered expression patterns for each cluster, visit that directory and run:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>$TRINITY_HOME/Analysis/DifferentialExpression/plot_expression_patterns.pl subcluster_*</code></pre>
</div></div>
<div class="paragraph"><p>This will generate a summary image file: my_cluster_plots.pdf, as shown below:</p></div>
<div class="paragraph"><p><span class="image">
<img src="../images/diff_expr/expression_profiles_for_clusters.png" alt="expression_profiles_for_clusters" />
</span></p></div>
</div>
<div class="sect2">
<h3 id="manually_define_clusters">Manually Defining Gene Clusters</h3>
<div class="paragraph"><p>Manually defining your clusters is the best way to organize the data to your liking. This is an interactive process. Fire up R from within your output directory, being sure it contains the <em>all.RData</em> file, and enter the following commands:</p></div>
<div class="literalblock">
<div class="content">
<pre><code>% R</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>&gt; load("all.RData") # check for your corresponding .RData file name to use here, replace all.RData accordingly</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>&gt; source("$TRINITY_HOME/Analysis/DifferentialExpression/R/manually_define_clusters.R")</code></pre>
</div></div>
<div class="literalblock">
<div class="content">
<pre><code>&gt; manually_define_clusters(hc_genes, data)</code></pre>
</div></div>
<div class="paragraph"><p>This should yield a display containing the hierarchically clustered genes, as shown below:</p></div>
<div class="paragraph"><p><span class="image">
<img src="../images/diff_expr/hcl_genes_tree.png" alt="expression_hcl_tree" />
</span></p></div>
<div class="paragraph"><p>Now, manually define your clusters from left to right (order matters here, so you can decipher the results later!) by clicking on the branch vertical branch that defines the clade of interest. After clicking on the branch, it will be drawn with a red box around the selected clade, as shown below:</p></div>
<div class="paragraph"><p><span class="image">
<img src="../images/diff_expr/manually_selected_clusters.png" alt="manually_selected_hcl_clusters_from_tree" />
</span></p></div>
<div class="paragraph"><p>Right click with the mouse (or double-touch a touchpad) to exit from cluster selection.</p></div>
<div class="paragraph"><p>The clusters as selected will be written to a subdirectory manually_defined_clusters_$count_clusters, and exist in a format similar to the automated-selection of clusters described above. Likewise, you can generate plots of the expression patterns for each cluster using the <em>plot_expression_patterns.pl</em> script.</p></div>
<div class="admonitionblock">
<table><tr>
<td class="icon">
<div class="title">Note</div>
</td>
<td class="content">The TMM-normalized FPKM matrix file can be imported into the <a href="http://www.tm4.org/mev.html">MeV MultiExperiment Viewer</a>, which provides sophisticated interactive visualization and analysis tools.  In addition, <a href="http://trinotate.sourceforge.net/TrinotateWeb.html">TrinotateWeb</a> is currently under development to provide interactive visualization methods for exploring transcriptome expression and annotation data.</td>
</tr></table>
</div>
</div>
</div>
</div>
</div>
<div id="footnotes"><hr /></div>
<div id="footer">
<div id="footer-text">
Last updated 2014-03-16 20:02:10 EDT
</div>
</div>
</body>
</html>
